lib/ansible/modules/cloud/amazon/iam_role.py

```diff
@@ -324,6 +324,20 @@ def create_or_update_role(connection, module):
             if not module.check_mode:
                 connection.add_role_to_instance_profile(InstanceProfileName=params['RoleName'], RoleName=params['RoleName'])
 
+    # Check Description update
+    if not module.check_mode and params.get('Description') and role['Description'] != params['Description']:
+        try:
+            if not module.check_mode:
+                connection.update_role_description(RoleName=params['RoleName'], Description=params['Description'])
+
+                changed = True
```

Comment:
I think this should be setting `changed = True` outside the `if` so that check mode will also show changed.